File: design-patterns/go/factory/transport/transport_factory.go
// transport_factory.go

package main

import (
	"errors"
	"strings"
)

type TransportFactory struct {
}

func NewTransportFactory() (transportFactory *TransportFactory) {
	transportFactory = &TransportFactory{}
	return
}

func (transportFactory *TransportFactory) GetTransport(identifier string) (Transport, error) {
	if strings.ToLower(identifier) == "bike" {
		return NewBike(), nil
	}

	if strings.ToLower(identifier) == "car" {
		return NewCar(), nil
	}

	if strings.ToLower(identifier) == "plane" {
		return NewPlane(), nil
	}

	return nil, errors.New("unknown transport identifier")
}

File: design-patterns/php/factory/transport/TransportFactory.php
<?php
// TransportFactory.php

namespace BigBoxCode\DesignPattern\Factory\Transport;

class TransportFactory {

    public static function getTransport(string $type): ?Transport {
        switch (strtolower($type)) {
            case "bike":
                return new Bike();
            case "car":
                return new Car();
            case "plane":
                return new Plane();
        }

        return null;
    }
}